File: app/src/main/java/de/karzek/diettracker/presentation/search/recipe/recipeEditDetails/adapter/viewHolder/RecipeEditDetailsMealSelectorViewHolder.java
package de.karzek.diettracker.presentation.search.recipe.recipeEditDetails.adapter.viewHolder;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Spinner;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnItemSelected;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.model.MealDisplayModel;
import de.karzek.diettracker.presentation.search.recipe.recipeEditDetails.adapter.itemWrapper.RecipeEditDetailsViewItemWrapper;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class RecipeEditDetailsMealSelectorViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.spinner_meal)
    Spinner spinner;

    private OnMealItemSelectedListener onMealItemSelectedListener;

    private Context context;

    public RecipeEditDetailsMealSelectorViewHolder(ViewGroup viewGroup,
                                                   OnMealItemSelectedListener onMealItemSelectedListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.viewholder_recipe_edit_details_meal_selector, viewGroup, false));
        ButterKnife.bind(this, itemView);

        context = viewGroup.getContext();
        this.onMealItemSelectedListener = onMealItemSelectedListener;
    }

    public void bind(RecipeEditDetailsViewItemWrapper item) {
        ArrayList<String> mealNames = new ArrayList<>();
        for (MealDisplayModel meal : item.getMeals())
            mealNames.add(meal.getName());

        ArrayAdapter<String> mealAdapter = new ArrayAdapter<>(context, android.R.layout.simple_spinner_item, mealNames);
        mealAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);

        spinner.setAdapter(mealAdapter);
        spinner.setSelection(item.getSelectedMeal());
    }

    @OnItemSelected(R.id.spinner_meal) public void onItemSelected(){
        onMealItemSelectedListener.onMealItemSelected(spinner.getSelectedItemPosition());
    }

    public interface OnMealItemSelectedListener {
        void onMealItemSelected(int id);
    }

}

File: app/src/main/java/de/karzek/diettracker/presentation/search/recipe/recipeEditDetails/adapter/viewHolder/RecipeEditDetailsMealSelectorViewHolder.kt
package de.karzek.diettracker.presentation.search.recipe.recipeEditDetails.adapter.viewHolder

import android.view.View
import android.widget.ArrayAdapter
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.presentation.search.recipe.recipeEditDetails.adapter.itemWrapper.RecipeEditDetailsViewItemWrapper
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.viewholder_recipe_edit_details_meal_selector.*
import java.util.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
class RecipeEditDetailsMealSelectorViewHolder(override val containerView: View,
                                              private val onMealItemSelectedListener: OnMealItemSelectedListener) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(item: RecipeEditDetailsViewItemWrapper) {
        val mealNames = ArrayList<String>()
        for (meal in item.meals)
            mealNames.add(meal.name)

        val mealAdapter = ArrayAdapter(containerView.context, android.R.layout.simple_spinner_item, mealNames)
        mealAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)

        spinner_meal.adapter = mealAdapter
        spinner_meal.setSelection(item.selectedMeal)
        spinner_meal.setOnClickListener { onMealItemSelectedListener.onMealItemSelected(spinner_meal.selectedItemPosition) }
    }

    interface OnMealItemSelectedListener {
        fun onMealItemSelected(id: Int)
    }

}
